### DISCUSSION TITLE
posts/reed-solomon-decoding

### DISCUSSION THREAD

**Discussion by giscus** [acciochris/acciochris.github.io#37] [Announcements]
# posts/reed-solomon-decoding

https://acciochris.github.io/posts/reed-solomon-decoding.html

**Comment by acciochris** [comment]
For some mobile users, the note about **Addition in $GF[2^3]$** is partially cut off, specifically the equations involving $p(x)$. The correct equations are

$$
\begin{align*}
  p(x) &= 0 \\
  x^3 + x + 1 &= 0 \\
  x^3 + x + 1 + x + 1 &= x + 1 \\
  x^3 &= x + 1
\end{align*}
$$

**Reply by acciochris** [reply]
I'm very sorry for the confusion incurred.

**Comment by acciochris** [comment]
As my professor has pointed out, there's a minor inconsistency between my example and my derivation regarding the last step of decoding.

Observe that as

$$
\Delta(x) = \prod_{j=1}^t (1 - X_jx)
$$

$\Delta(x)$ should theoretically be in the form $Ax^t + \cdots + 1$.

However, at the very end, I claimed that

$$
\Delta(x) = x + 3
$$

which is inconsistent with the fact that the constant portion should be $1$.

The way to resolve this problem is to multiply both $\Omega(x)$ and $\Delta(x)$ by $2$, the multiplicative inverse of $3$:

$$
\begin{align*}
  \Omega(x) & = 2 \times 2 = 3 \\
  \Delta(x) & = 2(x + 3) = 2x + 1 \\
\end{align*}
$$

However, this does not change the final result of the computation. Confirming this is left as an $\mathbf{Exercise}$ to the reader.
